test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/CSharpStatementTest/CatchClause.stree.txt

```diff
@@ -1,46 +1,46 @@
-Statement block - Gen<None> - 94 - (0:0,0)
-    Transition span - Gen<None> - [@] - SpanEditHandler;Accepts:None - (0:0,0) - Tokens:1
-        SyntaxKind.Transition;[@];
-    Code span - Gen<Stmt> - [try { foo(); } catch(IOException ioex) { handleIO(); } catch(Exception ex) { handleOther(); }] - SpanEditHandler;Accepts:Any - (1:0,1) - Tokens:42
-        SyntaxKind.Keyword;[try];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.LeftBrace;[{];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Identifier;[foo];
-        SyntaxKind.LeftParenthesis;[(];
-        SyntaxKind.RightParenthesis;[)];
-        SyntaxKind.Semicolon;[;];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.RightBrace;[}];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Keyword;[catch];
-        SyntaxKind.LeftParenthesis;[(];
-        SyntaxKind.Identifier;[IOException];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Identifier;[ioex];
-        SyntaxKind.RightParenthesis;[)];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.LeftBrace;[{];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Identifier;[handleIO];
-        SyntaxKind.LeftParenthesis;[(];
-        SyntaxKind.RightParenthesis;[)];
-        SyntaxKind.Semicolon;[;];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.RightBrace;[}];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Keyword;[catch];
-        SyntaxKind.LeftParenthesis;[(];
-        SyntaxKind.Identifier;[Exception];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Identifier;[ex];
-        SyntaxKind.RightParenthesis;[)];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.LeftBrace;[{];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Identifier;[handleOther];
-        SyntaxKind.LeftParenthesis;[(];
-        SyntaxKind.RightParenthesis;[)];
-        SyntaxKind.Semicolon;[;];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.RightBrace;[}];
+CSharpCodeBlock - [0..94)::94 - [@try { foo(); } catch(IOException ioex) { handleIO(); } catch(Exception ex) { handleOther(); }]
+    CSharpTransition - [0..1)::1 - Gen<None> - SpanEditHandler;Accepts:None
+        Transition;[@];
+    CSharpStatementLiteral - [1..94)::93 - [try { foo(); } catch(IOException ioex) { handleIO(); } catch(Exception ex) { handleOther(); }] - Gen<Stmt> - SpanEditHandler;Accepts:Any
```

Comment:
Lets discuss this next chance we get.

Reply:
@ajaybhargavb I actually think I was wrong here. After seeing more of this flesh out and seeing how we've been utilizing C# tokens as children I think what you have here is mostly fine.

I say "mostly fine" because I feel like `StatementLiteral` is too generic. In an ideal world it'd be best if we could differentiate `if` statements from `try` statements from `@{` blocks etc. That way when C# comes up with new language features (or we change our own) we can better define the impact on the syntax tree. A great example of this is the addition of `foreach await` in C# 8; it'd be awesome if we had `foreach` statement blocks that had a more defined structure that we could manipulate.

Anyhow, we can discuss tomorrow.